// react-hanger/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useWindowSize = exports.useToggleBodyClass = exports.useStateful = exports.useSetState = exports.useSet = exports.useScript = exports.usePrevious = exports.usePersist = exports.usePageLoad = exports.useOnClick = exports.useNumber = exports.useMap = exports.useLogger = exports.useInput = exports.useImage = exports.useGoogleAnalytics = exports.useFocus = exports.useDocumentReady = exports.useDelay = exports.useBoolean = exports.useArray = exports.useOnClickOutside = void 0;
var useClickOutside_1 = require("./useClickOutside");
Object.defineProperty(exports, "useOnClickOutside", { enumerable: true, get: function () { return useClickOutside_1.useOnClickOutside; } });
var useArray_1 = require("./useArray");
Object.defineProperty(exports, "useArray", { enumerable: true, get: function () { return useArray_1.useArray; } });
var useBoolean_1 = require("./useBoolean");
Object.defineProperty(exports, "useBoolean", { enumerable: true, get: function () { return useBoolean_1.useBoolean; } });
var useDelay_1 = require("./useDelay");
Object.defineProperty(exports, "useDelay", { enumerable: true, get: function () { return useDelay_1.useDelay; } });
var useDocumentReady_1 = require("./useDocumentReady");
Object.defineProperty(exports, "useDocumentReady", { enumerable: true, get: function () { return useDocumentReady_1.useDocumentReady; } });
var useFocus_1 = require("./useFocus");
Object.defineProperty(exports, "useFocus", { enumerable: true, get: function () { return useFocus_1.useFocus; } });
var useGoogleAnalytics_1 = require("./useGoogleAnalytics");
Object.defineProperty(exports, "useGoogleAnalytics", { enumerable: true, get: function () { return useGoogleAnalytics_1.useGoogleAnalytics; } });
var useImage_1 = require("./useImage");
Object.defineProperty(exports, "useImage", { enumerable: true, get: function () { return useImage_1.useImage; } });
var useInput_1 = require("./useInput");
Object.defineProperty(exports, "useInput", { enumerable: true, get: function () { return useInput_1.useInput; } });
var useLogger_1 = require("./useLogger");
Object.defineProperty(exports, "useLogger", { enumerable: true, get: function () { return useLogger_1.useLogger; } });
var useMap_1 = require("./useMap");
Object.defineProperty(exports, "useMap", { enumerable: true, get: function () { return useMap_1.useMap; } });
var useNumber_1 = require("./useNumber");
Object.defineProperty(exports, "useNumber", { enumerable: true, get: function () { return useNumber_1.useNumber; } });
var useOnClick_1 = require("./useOnClick");
Object.defineProperty(exports, "useOnClick", { enumerable: true, get: function () { return useOnClick_1.useOnClick; } });
var usePageLoad_1 = require("./usePageLoad");
Object.defineProperty(exports, "usePageLoad", { enumerable: true, get: function () { return usePageLoad_1.usePageLoad; } });
var usePersist_1 = require("./usePersist");
Object.defineProperty(exports, "usePersist", { enumerable: true, get: function () { return usePersist_1.usePersist; } });
var usePrevious_1 = require("./usePrevious");
Object.defineProperty(exports, "usePrevious", { enumerable: true, get: function () { return usePrevious_1.usePrevious; } });
var useScript_1 = require("./useScript");
Object.defineProperty(exports, "useScript", { enumerable: true, get: function () { return useScript_1.useScript; } });
var useSet_1 = require("./useSet");
Object.defineProperty(exports, "useSet", { enumerable: true, get: function () { return useSet_1.useSet; } });
var useSetState_1 = require("./useSetState");
Object.defineProperty(exports, "useSetState", { enumerable: true, get: function () { return useSetState_1.useSetState; } });
var useStateful_1 = require("./useStateful");
Object.defineProperty(exports, "useStateful", { enumerable: true, get: function () { return useStateful_1.useStateful; } });
var useToggleBodyClass_1 = require("./useToggleBodyClass");
Object.defineProperty(exports, "useToggleBodyClass", { enumerable: true, get: function () { return useToggleBodyClass_1.useToggleBodyClass; } });
var useWindowSize_1 = require("./useWindowSize");
Object.defineProperty(exports, "useWindowSize", { enumerable: true, get: function () { return useWindowSize_1.useWindowSize; } });
